aten/src/ATen/native/quantized/fake_quant_per_channel_affine.cpp

```diff
@@ -55,20 +59,19 @@ Tensor fake_quantize_per_channel_affine(
       "`axis` must be between 0 and number of dimensions of input");
 
   auto Y = at::empty_like(self, self.options(), MemoryFormat::Preserve);
-  for (int i = 0; i < self.size(axis); i++) {
-    auto input_slice = self.slice(axis, i, i + 1);
-    auto output_slice = Y.slice(axis, i, i + 1);
-    float sc = scale[i].item().toFloat();
-    int64_t z_point = zero_point[i].item().toLong();
-    fake_quant_slice_stub(
-        self.device().type(),
-        output_slice,
-        input_slice,
-        sc,
-        z_point,
-        quant_min,
-        quant_max);
-  }
+
+  std::vector<int64_t> expected_shape(self.dim(), 1);
+  expected_shape[axis] = self.size(axis);
+
+  TensorIterator iter;
+  iter.dont_compute_common_dtype();
+  iter.add_output(Y);
+  iter.add_input(self);
+  iter.add_input(scale.reshape(expected_shape));
```

Comment:
nit: using `view` here is slightly faster, and is guaranteed to work because input is 1d tensor.

Reply:
fixed